IoTuring/Entity/Deployments/Wifi/Wifi.py

```diff
@@ -0,0 +1,315 @@
+                    ValueFormatterOptions.TYPE_RADIOPOWER
+                ),
+            )
+        )
+
+    def Update(self):
+        p = self.RunCommand(self.commands[self.platform])
+        if not p.stdout:
+            raise Exception("error in GetWirelessInfo\n", p.stderr)
+        wifiInfo = self.GetWirelessInfo(p.stdout)
+        if not wifiInfo:
+            raise Exception("error while parsing wirelessInfo")
+        # set signal strength
+        if "Signal" in wifiInfo:  # Windows
+            self.SetEntitySensorValue(
+                key=self.keySignalStrength, value=wifiInfo["Signal"]
+            )
+        elif "Signal_Level" in wifiInfo:  # Linux
+            self.SetEntitySensorValue(
+                key=self.keySignalStrength, value=wifiInfo["Signal_Level"]
+            )
+        elif "agrCtlRSSI" in wifiInfo:
+            self.SetEntitySensorValue(
+                key=self.keySignalStrength, value=wifiInfo["agrCtlRSSI"]
+            )
+        else: # if there is no signal level found the interface might not be connected to an access point
+            self.SetEntitySensorValue(
+                key=self.keySignalStrength, value="not connected"
+            )
+        for key in self.patterns[self.platform]:
+            extraKey = "EXTRA_KEY_" + key.upper().replace(" ", "_")
+            self.SetEntitySensorExtraAttribute(
+                sensorDataKey=self.keySignalStrength,
+                attributeKey=globals()[extraKey],
+                attributeValue=wifiInfo[key] if key in wifiInfo else "not available",
+            )
+
+    def GetWirelessInfo(self, stdout):
+        wifi_info = {}
+        for key, pattern in self.patterns[self.platform].items():
+            match = re.search(pattern, stdout, re.IGNORECASE)
+            if match:
+                wifi_info[key] = match.group(1) if match.group(1) else match.group(0)
+
+        self.Log(self.LOG_DEBUG, wifi_info)
+        return wifi_info
+
+    @classmethod
+    def ConfigurationPreset(cls) -> MenuPreset:         
+
+        preset = MenuPreset()
+        preset.AddEntry(
+            name="Interface to check",
+            key=CONFIG_KEY_WIFI,
+            mandatory=True,
+            question_type="select",
+            choices=NIC_CHOICES
+        )
+        return preset
+
+    @classmethod
+    def CheckSystemSupport(cls):
+
+        def appendNicChoice(interfaceName, nicip4 = "", nicip6 = "", nicmac = ""):
+            NIC_CHOICES.append(
+                    {
+                        "name": WIFI_CHOICE_STRING.format(
+                            interfaceName,
+                            nicip4 if nicip4 else nicip6,  # defaults to showing ipv4
+                            nicmac,
+                        ),
+                        "value": interfaceName,
+                    }
+                )
+
+        ip4 = ""
+        ip6 = ""
+        mac = ""
+        interfaces = psutil.net_if_addrs()
+
```

Comment:
Since i'm checking for an wireless nic here, it would make sense to carry that over to configuration.
How do i carry this information over, in order to use it in ConfigurationPreset?

Reply:
Create a staticmethod or classmethod outside this, which returns the string/whatever you need. Than you can call that method twice, from both here and from ConfigurationPreset. (Or, you could assign it to a class parameter, but creating one plus method is easier, more readable, and similar to other code in this project)